[CARBONDATA-328] Improve Code and Fix Warnings [Spark]

PKOfficial opened the pull request:
Removed some compliation warnings.
Replace pattern matching for boolean to IF-ELSE.
Improved code according to scala standards.

**Please provide details on** 
**\- Whether new unit test cases have been added or why no new tests are required?**
Not required because no change in functionality.
**\- What manual testing you have done?**
Run basic Commands in Beeline.
**\- Any additional information to help reviewers in testing this change.**
No

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala:
```diff
@@ -259,29 +259,29 @@ object CarbonFilters {
+          if (list.exists(x => isNullLiteral(x.asInstanceOf[Literal]))) {
+            Some(new FalseExpression(transformExpression(a).get))
+          }
+          else {
```
put to previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/CarbonFilters.scala:
```diff
@@ -259,29 +259,29 @@ object CarbonFilters {
+          if (list.exists(x => isNullLiteral(x.asInstanceOf[Literal]))) {
+            Some(new FalseExpression(transformExpression(a).get))
+          }
+          else {
```
put to previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataLoadRDD.scala:
```diff
@@ -196,30 +196,29 @@ class DataFileLoaderRDD[K, V](
+        splits = CarbonQueryUtil.getTableSplitsForDirectLoad(carbonLoadModel.getFactFilePath,
+          partitioner.nodeList, partitioner.partitionCount)
+      }
+      else {
```
put to previous line, please modify all places

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataLoadRDD.scala:
```diff
@@ -439,42 +436,42 @@ class DataFileLoaderRDD[K, V](
+      logInfo("Preferred Location for split : " + firstOptionLocation.head)
+      val blockMap = new util.LinkedHashMap[String, Integer]()
+      val tableBlocks = theSplit.blocksDetails
+      tableBlocks.foreach(tableBlock => tableBlock.getLocations.foreach(
```
move `tableBlock.getLocations.foreach` to next line, and give proper indentation.
change `(` to `{` after foreach. the code standard is:
```
xxx.foreach { x => 
  x.yyy
}
```

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataLoadRDD.scala:
```diff
@@ -439,42 +436,42 @@ class DataFileLoaderRDD[K, V](
         }
-        )
+      )
+      )
```
indentation is not correct

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -175,8 +177,12 @@ object CarbonDataRDDFactory extends Logging {
   def configSplitMaxSize(context: SparkContext, filePaths: String,
-    hadoopConfiguration: Configuration): Unit = {
-    val defaultParallelism = if (context.defaultParallelism < 1) 1 else context.defaultParallelism
+      hadoopConfiguration: Configuration): Unit = {
```
in this case, write in one line is ok, no need to modify. better approach is to use min function instead of if else

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -189,15 +195,15 @@ object CarbonDataRDDFactory extends Logging {
       hadoopConfiguration.set(FileInputFormat.SPLIT_MAXSIZE, newSplitSize.toString)
       logInfo("totalInputSpaceConsumed : " + spaceConsumed +
-        " , defaultParallelism : " + defaultParallelism)
+              " , defaultParallelism : " + defaultParallelism)
```
change to `s"total... $spaceConsumed ... "` instead of string concat

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -505,129 +512,129 @@ object CarbonDataRDDFactory extends Logging {
-        override def run(): Unit = {
+    val compactionThread = new Thread {
+      override def run(): Unit = {
 
```
remove empty line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -505,129 +512,129 @@ object CarbonDataRDDFactory extends Logging {
+            )
+            triggeredCompactionStatus = true
+          }
+          catch {
```
move to previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -505,129 +512,129 @@ object CarbonDataRDDFactory extends Logging {
+                // not handling the exception. only logging as this is not the table triggered
+                // by user.
               }
+              finally {
```
previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -505,129 +512,129 @@ object CarbonDataRDDFactory extends Logging {
+                  logger.error("Exception in compaction thread for table " + tableForCompaction
+                    .carbonTable.getDatabaseName + "." +
+                               tableForCompaction.carbonTableIdentifier
+                                 .getTableName)
```
use `s" "` instead of string concat

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -505,129 +512,129 @@ object CarbonDataRDDFactory extends Logging {
-                  .tablesMeta.toArray, skipCompactionTables.toList.asJava
+            while (null != tableForCompaction) {
+              logger
+                .info("Compaction request has been identified for table " + tableForCompaction
```
please give proper format of log message. please modify all places

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -505,129 +512,129 @@ object CarbonDataRDDFactory extends Logging {
+              val tableCreationTime = CarbonEnv.getInstance(sqlContext).carbonCatalog
+                .getTableCreationTime(newCarbonLoadModel.getDatabaseName,
+                  newCarbonLoadModel.getTableName
+                )
```
move to previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -1043,32 +1050,32 @@ object CarbonDataRDDFactory extends Logging {
+        val metadataDetails = status(0)._2
+        if (!isAgg) {
+          val status = CarbonLoaderUtil
+            .recordLoadMetadata(currentLoadCount,
```
move to previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -1114,36 +1121,36 @@ object CarbonDataRDDFactory extends Logging {
-            .readLoadMetadata(loadMetadataFilePath)
+            // read latest table status again.
+            val latestMetadata = segmentStatusManager
+              .readLoadMetadata(loadMetadataFilePath)
```
move to previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -1114,36 +1121,36 @@ object CarbonDataRDDFactory extends Logging {
 
-          val latestStatus = CarbonLoaderUtil
-            .updateLoadMetadataFromOldToNew(details, latestMetadata)
+            val latestStatus = CarbonLoaderUtil
```
in this case, move `CarbonLoaderUtil` to next line, do not break the line between object and function

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala:
```diff
@@ -385,22 +385,22 @@ class CarbonGlobalDictionaryGenerateRDD(
           dictWriteTask.updateMetaData()
           // clear the value buffer after writing dictionary data
           valuesBuffer.clear
           org.apache.carbondata.core.util.CarbonUtil
```
remove package name

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala:
```diff
@@ -409,10 +409,10 @@ class CarbonGlobalDictionaryGenerateRDD(
       } finally {
         if (!dictionaryForDistinctValueLookUpCleared) {
           org.apache.carbondata.core.util.CarbonUtil
```
remove package name, modify all places

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala:
```diff
@@ -49,7 +49,7 @@ object CommonUtil {
       }
       // if invalid column is present
-      else if (dims.filter { dim => dim.column.equalsIgnoreCase(x) }.isEmpty) {
+      else if (!dims.exists { dim => dim.column.equalsIgnoreCase(x) }) {
```
use `!dims.exists(_.column.equalsIgnoreCase(x))` instead of having `{` 
modify all places

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala:
```diff
@@ -267,10 +267,10 @@ object GlobalDictionaryUtil extends Logging {
-                              model: DictionaryLoadModel): Boolean = {
+      rowCount: Long,
+      model: DictionaryLoadModel): Boolean = {
     (columnCardinality > model.highCardThreshold) && (rowCount > 0) &&
```
move `(rowCount > 0)` to next line so that every condition is one line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala:
```diff
@@ -413,12 +413,12 @@ object GlobalDictionaryUtil extends Logging {
-          model.table.getTableName)(sqlContext)
-        .asInstanceOf[CarbonRelation].tableMeta.carbonTable
+      model.table.getTableName)(sqlContext)
+      .asInstanceOf[CarbonRelation].tableMeta.carbonTable
```
move to previous line if not exceeding 100 characters

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala:
```diff
@@ -839,10 +841,10 @@ object GlobalDictionaryUtil extends Logging {
           // prune columns according to the CSV file header, dimension columns
           val (requireDimension, requireColumnNames) =
-            pruneDimensions(dimensions, headers, headers)
+          pruneDimensions(dimensions, headers, headers)
```
incorrect indentation

jackylk commented:
Thanks for working for this

PKOfficial commented:
I've fixed code style and changed string concatenation to string interpolation.
Please have a look and let me know if you want more changes.
Thanks.

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataLoadRDD.scala:
```diff
@@ -196,30 +195,28 @@ class DataFileLoaderRDD[K, V](
-        blocksGroupBy.zipWithIndex.map{b =>
-          new CarbonNodePartition(id, b._2, b._1._1, b._1._2)
-        }
+      splits.zipWithIndex.map { s =>
```
instead of `s`, please give a more meaningful name, like `case (split, index)`

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -188,16 +194,16 @@ object CarbonDataRDDFactory extends Logging {
-      logInfo("totalInputSpaceConsumed : " + spaceConsumed +
-        " , defaultParallelism : " + defaultParallelism)
-      logInfo("mapreduce.input.fileinputformat.split.maxsize : " + newSplitSize.toString)
+      logInfo(s"totalInputSpaceConsumed : $spaceConsumed ," +
```
remove space before `:` and make this log in one line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -264,8 +270,8 @@ object CarbonDataRDDFactory extends Logging {
         logger
-          .info("Acquired the compaction lock for table " + carbonLoadModel
-            .getDatabaseName + "." + carbonLoadModel.getTableName
+          .info(s"Acquired the compaction lock for table ${ carbonLoadModel.getDatabaseName } " +
```
move to previous line

jackylk commented on integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala:
```diff
@@ -277,44 +283,43 @@ object CarbonDataRDDFactory extends Logging {
         logger
-          .error("Not able to acquire the compaction lock for table " + carbonLoadModel
-            .getDatabaseName + "." + carbonLoadModel.getTableName
+          .error(s"Not able to acquire the compaction lock for table" +
```
move to previous line

jackylk commented:
@PKOfficial are you working on this?

PKOfficial commented:
@jackylk Yes ,
Done with almost everything except meaningful names of variables, I am currently working on that and it would be completed by today.

jackylk commented:
please rebase.

PKOfficial commented:
@jackylk : I've rebased please have a look

jackylk commented:
I think there are still repeated commits, can you create a new branch based on latest master code and squash your commit?

PKOfficial commented:
@jackylk I've create the new PR with new branch, Please have a look
https://github.com/apache/incubator-carbondata/pull/279
